content/concepts/security-considerations.md

```diff
@@ -0,0 +1,140 @@
+password. This could be accomplished by defining an authorization protocol that
+accepts usernames and passwords and responds with a signed token if the
+credentials are valid. Protocols that expose sensitive resources could then
+require the token before allowing access.
+
+Systems that are designed to be fully decentralized are often "open by default,"
+allowing any peer to participate in core functions. However, such systems may
+benefit from maintaining some kind "reputation" system to identify faulty or
+malicious participants and block or ignore them. Implementing decentralized
+reputation management is outside the scope of libp2p, but many of libp2p's core
+developers and community members are excited by research and development in this
+area, and would welcome your thoughts [on the libp2p
+forums](https://discuss.libp2p.io).
+
+## Cooperative Systems with Abuse Potential
+
+Some of libp2p's most useful built-in protocols are cooperative, leveraging
+other peers in the network to perform tasks that benefit everyone. For example,
+data stored on the Kad-DHT is replicated across the set of peers that are
+"closest" to the data's associated key, whether those peers have any particular
+interest in the data or not.
+
+Cooperative systems are inherently succeptible to abuse by bad actors, and
+although we are researching ways to limit the impact of such attacks, they are
+possible in libp2p today.
+
+
+### Kad-DHT
+
+The Kad-DHT protocol is a [distributed hash table][glossary-dht] that provides a
+shared key/value storage system for all participants. In addition to key/value
+lookups, the DHT is the default implementation of libp2p's [peer
+routing][concepts-peer-routing] and [content routing][concepts-content-routing]
+interfaces, and thus serves an important role in discovering other peers and
+services on the network.
+
+#### Sybil Attacks
+
+DHTs in general are vulnerable to a class of attacks called [Sybil
+attacks][wikipedia-sybil], in which one operator spins up a large number of DHT
+peers (generally called "Sybils") to gain undue influence over the routing
+algorithm used to maintain the network.
+
+If many Sybils are generated with IDs that are "close" to a particular key in
+the DHT, they are likely to be in the path for most queries for that key. This
+gives them an opportunity to modify query responses, either by returning
+incorrect data or by not returning data at all.
+
+Applications can guard against modification of data by signing values that are
+stored in the DHT, or by using a hash of the data as the DHT key (as in
+[IPFS](https://ipfs.io)). These strategies allow you to detect if the data has
+been tampered with, however, they cannot prevent tampering from occuring in the
+first place, nor can they prevent malicious nodes from simply pretending the
+data doesn't exist and omitting it entirely.
+
+Very similar to Sybil attacks, an Eclipse attack also uses a large number of
```

Comment:
It requires the underlying network to attend to some inherent structure, which in the case of the Kademlia algorithm, is determined by the parameters of the routing table and the XOR distance function.